tuf/formats.py

```diff
@@ -230,33 +230,40 @@
 # Information about target files, like file length and file hash(es).  This
 # schema allows the storage of multiple hashes for the same file (e.g., sha256
 # and sha512 may be computed for the same file and stored).
-FILEINFO_SCHEMA = SCHEMA.Object(
-  object_name = 'FILEINFO_SCHEMA',
+TARGETS_FILEINFO_SCHEMA = SCHEMA.Object(
+  object_name = 'TARGETS_FILEINFO_SCHEMA',
   length = LENGTH_SCHEMA,
   hashes = HASHDICT_SCHEMA,
-  version = SCHEMA.Optional(METADATAVERSION_SCHEMA),
   custom = SCHEMA.Optional(SCHEMA.Object()))
 
+# Information about snapshot and timestamp files. This schema allows for optional
+# length and hashes, but version is mandatory.
+METADATA_FILEINFO_SCHEMA = SCHEMA.Object(
+  object_name = 'METADATA_FILEINFO_SCHEMA',
+  length = SCHEMA.Optional(LENGTH_SCHEMA),
+  hashes = SCHEMA.Optional(HASHDICT_SCHEMA),
+  version = METADATAVERSION_SCHEMA)
+
 # A dict holding the version or file information for a particular metadata
 # role.  The dict keys hold the relative file paths, and the dict values the
 # corresponding version numbers and/or file information.
 FILEINFODICT_SCHEMA = SCHEMA.DictOf(
   key_schema = RELPATH_SCHEMA,
   value_schema = SCHEMA.OneOf([VERSIONINFO_SCHEMA,
-                              FILEINFO_SCHEMA]))
+                              METADATA_FILEINFO_SCHEMA]))
 
 # A dict holding the information for a particular target / file.  The dict keys
 # hold the relative file paths, and the dict values the corresponding file
 # information.
 FILEDICT_SCHEMA = SCHEMA.DictOf(
   key_schema = RELPATH_SCHEMA,
-  value_schema = FILEINFO_SCHEMA)
+  value_schema = TARGETS_FILEINFO_SCHEMA)
 
 # A dict holding a target info.
 TARGETINFO_SCHEMA = SCHEMA.Object(
   object_name = 'TARGETINFO_SCHEMA',
   filepath = RELPATH_SCHEMA,
-  fileinfo = FILEINFO_SCHEMA)
+  fileinfo = TARGETS_FILEINFO_SCHEMA)
```

Comment:
After a good investigation, I found out that this would require a separate pr. 

`FILEDICT_SCHEMA` and `TARGETINFO_SCHEMA` are not actually 100 % the same.
FILEDICT_SCHEMA  is a dict in the format `{ "path" : "fileinfo" }` and
TARGETINFO_SCHEMA  is a dict in the format `{ "filepath" : <something> , "fileinfo" : <something>}`.

This could be seen pretty well into the `tests/test_formats.py`:
![image](https://user-images.githubusercontent.com/16246778/83659760-a1854a80-a5cc-11ea-8691-5ba24ae88e10.png)

Reply:
WIll track the progress here: https://github.com/theupdateframework/tuf/issues/1043